pkg/clusterapi/system.go

```diff
@@ -175,6 +176,25 @@ func (c *system) Run(ctx context.Context, installConfig *installconfig.InstallCo
 			),
 		)
 	case gcp.Name:
+		gAppCredEnvVar := "GOOGLE_APPLICATION_CREDENTIALS"
+		capgEnvVars := map[string]string{}
+		session, err := gcpic.GetSession(context.Background())
+		if err != nil {
+			return fmt.Errorf("failed to create gcp session: %w", err)
+		}
+
+		if session.Path != "" {
+			if _, err := os.Stat(session.Path); err == nil {
+				capgEnvVars[gAppCredEnvVar] = session.Path
+			}
+		}
+
+		if v, ok := capgEnvVars[gAppCredEnvVar]; ok {
+			logrus.Infof("setting %q to %s for capg infrastructure controller", gAppCredEnvVar, v)
+		} else {
+			return fmt.Errorf("failed to find valid gcp authentication data")
+		}
```

Comment:
In this case wouldn't users be using default application credentials? So we don't want to throw an error.